mitiq/pec/representations/tests/test_learning.py

```diff
@@ -199,54 +236,120 @@ def noisy_execute_qiskit(circ: qiskit.QuantumCircuit) -> float:
+            "./mitiq/pec/representations/tests/learning_pec_data",
+            f"learning_pec_data_eps_{eps_string}.txt",
+        )
+    )
 
     [success, epsilon_opt] = learn_depolarizing_noise_parameter(
         operations_to_learn=operations_to_learn,
         circuit=circuit,
         ideal_executor=ideal_executor,
         noisy_executor=noisy_executor,
-        pec_kwargs=pec_kwargs_learning,
         num_training_circuits=5,
         fraction_non_clifford=0.2,
         training_random_state=np.random.RandomState(1),
         epsilon0=epsilon0,
         observable=observable,
+        learning_kwargs={"pec_data": pec_data},
     )
     assert success
     assert abs(epsilon_opt - epsilon) < offset * epsilon
+
+
+@pytest.mark.parametrize("epsilon", [0.05, 0.1])
+@pytest.mark.parametrize("eta", [1, 2])
+def test_learn_biased_noise_parameters(epsilon, eta):
+    """Test the learning function can run without pre-executed data"""
+
+    operations_to_learn = [Circuit(op[1]) for op in CNOT_ops]
+
+    def noisy_execute(circ: Circuit) -> np.ndarray:
+        noisy_circ = circ.copy()
+        insertions = []
+        for op in CNOT_ops:
+            index = op[0] + 1
+            qubits = op[1].qubits
+            for q in qubits:
+                insertions.append(
+                    (index, biased_noise_channel(epsilon, eta)(q))
+                )
+        noisy_circ.batch_insert(insertions)
+        return ideal_execute(noisy_circ)
+
+    noisy_executor = Executor(noisy_execute)
+
+    eps_offset = 0.1
+    eta_offset = 0.2
+    epsilon0 = (1 - eps_offset) * epsilon
+    eta0 = (1 - eta_offset) * eta
+
+    num_training_circuits = 5
+    pec_data = np.zeros([122, 122, num_training_circuits])
+
+    eps_string = str(epsilon).replace(".", "_")
+    for tc in range(0, num_training_circuits):
+        pec_data[:, :, tc] = np.loadtxt(
+            os.path.join(
+                "./mitiq/pec/representations/tests/learning_pec_data",
+                f"learning_pec_data_eps_{eps_string}eta_{eta}tc_{tc}.txt",
+            )
+        )
+
+    [success, epsilon_opt, eta_opt] = learn_biased_noise_parameters(
+        operations_to_learn=operations_to_learn,
+        circuit=circuit,
+        ideal_executor=ideal_executor,
+        noisy_executor=noisy_executor,
+        num_training_circuits=num_training_circuits,
+        fraction_non_clifford=0.2,
+        training_random_state=np.random.RandomState(1),
+        epsilon0=epsilon0,
+        eta0=eta0,
+        observable=observable,
+        learning_kwargs={"pec_data": pec_data},
+    )
+    assert success
+    assert abs(epsilon_opt - epsilon) < eps_offset * epsilon
+    assert abs(eta_opt - eta) < eta_offset * eta
+
+
+learning_kwargs = {}
+
+
+def test_parse_learning_kwargs():
+    pec_data, method, minimize_kwargs = _parse_learning_kwargs(
+        learning_kwargs=learning_kwargs
+    )
+    assert pec_data is None
+    assert method == "Nelder-Mead"
+    assert minimize_kwargs == {}
```

Comment:
We do test what happens when `learning_kwargs` isn't empty- in `test_learn_biased_noise_parameters` and `test_learn_depolarizing_noise_parameters` we pass the data in that way. Do we still need an explicit case in `test_parse_learning_kwargs`?

Reply:
Ah okay I see `pec_data` is used in the other test, but neither `method` nor `minimize_kwargs` are tested. Is it worth testing for non-empty values for those? It might not be worth it. Fine with me either way.

If we keep this test as is, I would request it's renamed to reflect it's testing empty `learning_kwargs` and it's not a more general test.

Reply:
I think renaming to `test_empty_learning_kwargs` is the best option.